### PR TITLE
Enhancement global nav

### PR DESCRIPTION
**Pull request by KorirC**
### Description
Added ability to close app menu and user menu on the right Nav Bar,  when user clicks outside.

### CODE REVIEW

File: packages/apps/esm-primary-navigation-app/src/components/navbar-header-panels/app-menu-panel.component.tsx [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
 
 interface AppMenuProps {
   expanded: boolean;
+  hidePanel: () => void;
 }
 
-const AppMenuPanel: React.FC<AppMenuProps> = ({ expanded }) => {
+const AppMenuPanel: React.FC<AppMenuProps> = ({ expanded, hidePanel }) => {
+  const ref = useRef();
+  useOnClickOutside(ref, () => hidePanel());
```
Comment (FlorianRappl) [inline packages/apps/esm-primary-navigation-app/src/components/navbar-header-panels/app-menu-panel.component.tsx:14]: only `hidePanel` is sufficient, i.e., `useOnClickOutside(ref, hidePanel)`

File: packages/apps/esm-primary-navigation-app/src/components/navbar-header-panels/useOnClickOutside.component.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      if (!ref.current || ref.current.contains(event.target)) {
+        return;
+      }
+      handler(event);
+    };
+    document.addEventListener("click", listener);
+    return () => {
+      document.removeEventListener("click", listener);
+    };
+  }, [ref, handler]);
```
Comment (FlorianRappl) [inline packages/apps/esm-primary-navigation-app/src/components/navbar-header-panels/useOnClickOutside.component.tsx:15]: The `ref` is constant anyway, maybe you should look for the `ref.current`, but that does not fire when changed. Still, its better as a dependency as `ref`

File: packages/apps/esm-primary-navigation-app/src/components/navbar-header-panels/useOnClickOutside.component.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { useEffect } from "react";
+
+export function useOnClickOutside(ref, handler) {
```
Comment (FlorianRappl) [inline packages/apps/esm-primary-navigation-app/src/components/navbar-header-panels/useOnClickOutside.component.tsx:3]: Types?

File: packages/apps/esm-primary-navigation-app/src/components/navbar/navbar.component.tsx [hunk 4]
```diff
@@ -140,17 +140,25 @@ const Navbar: React.FC<NavbarProps> = ({
...
             expanded={isActivePanel("notificationsMenu")}
           />
-          <UserMenuPanel
-            user={user}
-            session={session}
-            expanded={isActivePanel("userMenu")}
-            allowedLocales={allowedLocales}
-            onLogout={onLogout}
-          />
+          {isActivePanel("userMenu") && (
```
Comment (FlorianRappl) [inline packages/apps/esm-primary-navigation-app/src/components/navbar/navbar.component.tsx:152]: Do we really need the "double" isActive?

Reason: The inner isActive will actually hide / show from carbon, just via CSS - while the outer will mount / unmount DOM nodes (inefficient).

**Review by FlorianRappl — COMMENTED** [review]
LGTM

File: packages/apps/esm-primary-navigation-app/src/components/navbar-header-panels/app-menu-panel.component.tsx [hunk 5]
```diff
@@ -1,15 +1,31 @@
...
+    if (expanded) {
+      const listener = (event: MouseEvent) => {
+        if (appMenuRef?.current && !appMenuRef.current.contains(event.target)) {
+          hidePanel();
+        }
+      };
+      window.addEventListener("click", listener);
+      return () => window.removeEventListener("click", listener);
+    }
+  }, [appMenuRef?.current]);
```
Comment (FlorianRappl) [inline packages/apps/esm-primary-navigation-app/src/components/navbar-header-panels/app-menu-panel.component.tsx:24]: Missing  `expanded`

File: packages/apps/esm-primary-navigation-app/src/components/navbar-header-panels/user-menu-panel.component.tsx [hunk 6]
```diff
@@ -18,9 +19,28 @@ const UserMenuPanel: React.FC<UserMenuPanelProps> = ({
...
+          userMenuRef?.current &&
+          !userMenuRef.current.contains(event.target)
+        ) {
+          hidePanel();
+        }
+      };
+      window.addEventListener("click", listener);
+      return () => window.removeEventListener("click", listener);
+    }
+  }, [userMenuRef?.current]);
```
Comment (FlorianRappl) [inline packages/apps/esm-primary-navigation-app/src/components/navbar-header-panels/user-menu-panel.component.tsx:39]: Missing `expanded` - the whole code is the same, why not reuse it with a custom hook?

**Review by donaldkibet — COMMENTED** [review]
@KorirC fix the lint error

**Comment by manuelroemer** [comment]
@KorirC By the way, you should be able to fix a lot of the linting errors by running `yarn prettier` in the repo's root directory (this automatically fixes any code styling issues).

**Review by donaldkibet — APPROVED** [review]
LTGM 🚀   thanks @KorirC

**Comment by gracepotma** [comment]
Eric Achillah reports that this feature is not working as expected even though it looks like it was merged. @donaldkibet it looks like what might have happened is that you only merged b2dd8ed? 
@KorirC can you restore the branch you deleted, so @donaldkibet can merge the whole thing? (Perhaps this will fix? @donaldkibet can correct me if I've misunderstood something)

**Pull request merged**